### DISCUSSION TITLE
Specifying IC's Over 2-D Domain

### DISCUSSION THREAD

**Discussion by jsevic** [idaholab/moose#26434] [Q&A Modules: Phase field] [answered]
I have found question #21209 asked about using a .csv file to specify IC's and I have a similar question.

Suppose I run a PF simulation to steady-state that yields some equilibrium concentration c(x,y). Now what I would like to do is take this equilibrium concentration and import back into MOOSE as the IC for concentration, c(x,y,0).

I see something called Polycrystal Initial Conditions also?

Is there a way to do this thx!

**Comment by jsevic** [comment]
I think I found it here...

https://mooseframework.inl.gov/syntax/ICs/index.html

**Reply by GiudGiud** [reply, marked as answer]
If it s coming from MOOSE into MOOSE, then we usually just use a SolutionUserObject with a SOlutionIC. No need for a CSV.

CSV will be helpful if you are trying to process the field in an intermediate step. ANd then PropertyReadFile / PiecewiseConstantFromCSV is what I use

**Reply by jsevic** [reply]
OK I will try thx!
